### PR TITLE
Add support for XGBoost flavor

### PR DESCRIPTION
**Pull request by harupy**
## What changes are proposed in this pull request?

Add support for XGBoost flavor 

## How is this patch tested?

Unit tests

### Is this a user-facing change?

- [ ] No. You can skip the rest of this section.
- [x] Yes. Give a description of this change to be included in the release notes for MLflow users.

Model API supports XGBoost.

### What component(s) does this PR affect?

- [ ] UI
- [ ] CLI
- [x] API
- [ ] REST-API
- [ ] Examples
- [ ] Docs
- [ ] Tracking
- [ ] Projects
- [ ] Artifacts
- [ ] Models
- [ ] Scoring
- [ ] Serving
- [ ] R
- [ ] Java
- [ ] Python

### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [ ] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [x] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

File: mlflow/xgboost.py [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+    """
+    :return: The default Conda environment for MLflow Models produced by calls to
+             :func:`save_model()` and :func:`log_model()`.
+    """
+    import xgboost as xgb
+
+    return _mlflow_conda_env(
+        additional_conda_deps=None,
+        additional_pip_deps=[
+            "xgboost=={}".format(xgb.__version__),
```
Comment (smurching) [inline mlflow/xgboost.py]: Ah interesting, surprising that xgboost isn't available on the default conda channels :P (verified this by running `conda install xgboost` on my Mac, which failed)

File: mlflow/xgboost.py [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+                            ]
+                        }
+
+    :param mlflow_model: :py:mod:`mlflow.models.Model` this flavor is being added to.
+    """
+    import xgboost as xgb
+
+    path = os.path.abspath(path)
+    if os.path.exists(path):
+        raise Exception("Path '{}' already exists".format(path))
```
Comment (smurching) [inline mlflow/xgboost.py]:
```suggestion
        raise MlflowException("Path '{}' already exists".format(path), DIRECTORY_NOT_EMPTY)
```

The suggestion is to raise the more specific MlflowException so users can catch that if they want (note that they can still catch a broader `Exception`)

**Review by smurching — COMMENTED** [review]
The implementation looks pretty good (barring small comments) :)

Would you mind adding tests for `log_model` and `save_model`? You can see  `tests/keras/test_keras_model_export.py` for an example to work from

File: mlflow/xgboost.py (hunk 1 above)
Comment (harupy) [reply mlflow/xgboost.py]: Did you try `conda install -c conda-forge xgboost`?
I always use this command to install xgboost on conda.

https://anaconda.org/conda-forge/xgboost

File: mlflow/xgboost.py (hunk 2 above)
Comment (harupy) [reply mlflow/xgboost.py]: will fix this

**Comment by harupy** [comment]
@smurching Thanks for the review! I'll add tests based on keras flavor.

**Review by smurching — APPROVED** [review]
LGTM thanks for the updates @harupy!

**Comment by harupy** [comment]
@smurching Thanks for merging! I'll start working on LightGBM flavor.

**Pull request merged**